app/controllers/stories_controller.rb

```diff
@@ -240,24 +240,35 @@ def assign_feed_stories
   end
 
   def assign_article_show_variables
+    not_found if permission_denied?
+    not_found unless @article.user
```

Comment:
right now these two are hidden in the rest of the method, let's bail out quickly if the conditions are not met